clients/src/main/java/org/apache/kafka/common/requests/ReadShareGroupStateSummaryRequest.java

```diff
@@ -83,8 +84,24 @@ public ReadShareGroupStateSummaryRequestData data() {
 
     public static ReadShareGroupStateSummaryRequest parse(ByteBuffer buffer, short version) {
         return new ReadShareGroupStateSummaryRequest(
-                new ReadShareGroupStateSummaryRequestData(new ByteBufferAccessor(buffer), version),
-                version
+            new ReadShareGroupStateSummaryRequestData(new ByteBufferAccessor(buffer), version),
+            version
         );
     }
+
+    public static List<ReadShareGroupStateSummaryResponseData.ReadStateSummaryResult> getErrorList(
+        List<Uuid> topicIds,
+        Errors error
+    ) {
+        return topicIds.stream()
+            .map(topicId ->
+                new ReadShareGroupStateSummaryResponseData.ReadStateSummaryResult()
+                    .setTopicId(topicId)
+                    .setPartitions(List.of(
+                            new ReadShareGroupStateSummaryResponseData.PartitionResult()
+                                .setErrorCode(error.code())
+                                .setErrorMessage(error.message())
```

Comment:
This will give default Error message and original exception message will be lost, do you want that behaviour? If yes, then please write a comment here else please fix it.